samples/sample_full_pbr.cpp

```diff
@@ -281,9 +281,7 @@ static void setup(Engine* engine, View* view, Scene* scene) {
             vec2 uvDy = dFdy(uv0);
 
             mat3 tangentFromWorld = transpose(getWorldTangentFrame());
-            vec3 tangentCameraPosition = tangentFromWorld * getWorldCameraPosition();
-            vec3 tangentFragPosition = tangentFromWorld * getWorldPosition();
-            vec3 v = normalize(tangentCameraPosition - tangentFragPosition);
+            vec3 v = tangentFromWorld * getWorldViewVector();
```

Comment:
Have you tested this change?

Reply:
I tried testing it but this sample doesn't work for me with the Metal backend (not even on main branch). I tried using assets/models/monkey/monkey.obj (with a height map generated in GIMP to test the parallax mapping code path).

Firstly, the following line gave me a compiler error in `MaterialBuilder`:
`for (int i = 0; i < numLayers; i++) {`

```metal
ERROR: 0:24: '<' :  wrong operand types: no operation '<' exists that takes a left-hand operand of type ' temp mediump int' and a right operand of type ' temp mediump float' (or there is no acceptable conversion)
ERROR: 0:24: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.
```

Fixing the error got me past the Precondition panic but the monkey was rendered flat black and upon rotating the camera the render target content got stuck (wasn't cleared) so I gave up.

But here `tangentFromWorld` is [the usual TBN matrix](https://github.com/google/filament/blob/main/shaders/src/shading_parameters.fs#L31) containing only rotation and mirroring. Thus it is safe to use it to transform direction vectors (no need to transform eye/fragment pos to tangent space). It also keeps the vector's length, so the normalizing at the end is not needed (because `getWorldViewVector()` is already normalized)

Reply:
This part can be reverted too if you'd like. I figured this fixes a bug, without modifying the public API, that's why I kept this

Reply:
Reverted in cec57142d6c13d658c3f859648f0471291ae46cb.

Reply:
https://github.com/google/filament/pull/6490